Java/src/main/java/net/hypixel/api/reply/PlayerReply.java

```diff
@@ -1,23 +1,358 @@
 package net.hypixel.api.reply;
 
+import com.google.gson.JsonArray;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
+import java.util.Date;
+import java.util.UUID;
+import net.hypixel.api.util.ILeveling;
 
 public class PlayerReply extends AbstractReply {
-    private JsonElement player;
 
-    public JsonObject getPlayer() {
-        if (player == null || player.isJsonNull()) {
-            return null;
-        } else {
-            return player.getAsJsonObject();
-        }
+    private static final Player BLANK_PLAYER = new Player();
+
+    private Player player;
+
+    public Player getPlayer() {
+        return player == null ? BLANK_PLAYER : player;
     }
 
     @Override
     public String toString() {
         return "PlayerReply{" +
-                "player=" + player +
-                "} " + super.toString();
+            "player=" + player +
+            "} " + super.toString();
+    }
+
+    public static class Player {
+
+        private static final String DEFAULT_RANK = "NONE";
+
+        private JsonElement raw;
+
+        /**
+         * @return A string representing the player's Minecraft UUID (without hyphens)
+         */
+        public String getUuid() {
+            return getStringProperty("uuid", null);
+        }
+
+        /**
+         * @return The Minecraft username the player had when they last connected to Hypixel
+         */
+        public String getName() {
+            return getStringProperty("displayname", null);
```

Comment:
This should first try to use `displayname`, if it's null, then last element of `knownAliases`, if there's none, then `playname`, and then `username`.

Reply:
Should we bother checking `playername` since its meant to be a lowercase variant? If we're not checking `knownAliasesLower` as well, I don't see the need to check this one. Do you know of any players who only have a `playername` and no `displayname`?

Reply:
Yes, there are players with no displayname. Not sure if we should check `playername` and `username`, but I guess lowercase name is better than nothing.
About `knownAliasesLower`, I think if there's no `knownAliases`, then there would be no `knownAliasesLower` too.